apps/Vision/HeatMatrix.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from queue import Queue
+
+class HeatMatrix:
+
+   def __init__(self, size, queue_size, origin=None):
+      if size <= 1:
+         raise ValueTooSmallError
+      
+      if origin is not None:
+         if origin[0] - size >= 0 or origin[1] - size >= 0:
+            raise InvalidOriginError
+         self.origin = origin
+      else:
+         self.origin = np.array([size // 2,size // 2]) 
+
+      self.shape = (size,size)
+      self.queue_size = queue_size
+      self.frames_queue = Queue(self.queue_size)
+      self.heat_matrix = np.ones(self.shape)
+      self._init_queue()
+
+   def _get_oldest_frame(self):
+      oldest_frame = self.frames_queue.get()
+      return oldest_frame
+
+   def _add_new_frame(self, frame):
+      self.frames_queue.put(frame)
+
+   def _init_queue(self):
+      dummy_frame = np.ones(self.shape)
+      for q in range(self.queue_size):
+         self.frames_queue.put(dummy_frame)
+
+   def _update_heat_matrix(self, new_frame):
+
+      #Compute heat matrix by by removing data from the last frame (divide)
+      #and add new frame (multiply)
+      self.heat_matrix = (self.heat_matrix / self._get_oldest_frame()) * new_frame
```

Comment:
I could be wrong here, but I feel we will want to more so decrease the intensity of how much each frame is contributing to the matrix, for example everytime the update heat matrix is called maybe we can multiply each frame by 1/2 so that frame 5 is only contributing 1/16 of its original value, frame 4 - 1/8th, frame 3 - 1/4, frame 2 - 1/2 and the new frame (frame 1) is contributing its full value. Let me know if that makes sense to you... I could be off in how I am picturing the functionality of this function.